File: test_geographic_data.py
#!/usr/bin/env python3
"""
Test des données géographiques
"""

import requests
import json

# Configuration
BASE_URL = "http://127.0.0.1:8000/api"
GEOGRAPHIC_URL = f"{BASE_URL}/users/geographic-data/"

def test_geographic_data():
    """Test des données géographiques"""
    
    print("🗺️ TEST DES DONNÉES GÉOGRAPHIQUES")
    print("=" * 50)
    
    try:
        response = requests.get(GEOGRAPHIC_URL)
        print(f"Status: {response.status_code}")
        
        if response.status_code == 200:
            data = response.json()
            print("✅ Données géographiques récupérées avec succès")
            
            # Analyser la structure
            regions = data.get('regions', [])
            print(f"\n📊 STATISTIQUES:")
            print(f"   - Régions: {len(regions)}")
            
            total_prefectures = 0
            total_communes = 0
            total_quartiers = 0
            
            for region in regions:
                prefectures = region.get('prefectures', [])
                total_prefectures += len(prefectures)
                
                for prefecture in prefectures:
                    communes = prefecture.get('communes', [])
                    total_communes += len(communes)
                    
                    for commune in communes:
                        quartiers = commune.get('quartiers', [])
                        total_quartiers += len(quartiers)
            
            print(f"   - Préfectures: {total_prefectures}")
            print(f"   - Communes: {total_communes}")
            print(f"   - Quartiers: {total_quartiers}")
            
            # Afficher quelques exemples
            print(f"\n🏛️ EXEMPLES DE RÉGIONS:")
            for i, region in enumerate(regions[:3]):  # Afficher les 3 premières
                print(f"   {i+1}. {region.get('nom', 'N/A')} ({region.get('code', 'N/A')})")
                
                prefectures = region.get('prefectures', [])
                if prefectures:
                    prefecture = prefectures[0]
                    print(f"      - Préfecture: {prefecture.get('nom', 'N/A')}")
                    
                    communes = prefecture.get('communes', [])
                    if communes:
                        commune = communes[0]
                        print(f"        - Commune: {commune.get('nom', 'N/A')}")
                        
                        quartiers = commune.get('quartiers', [])
                        if quartiers:
                            quartier = quartiers[0]
                            print(f"          - Quartier: {quartier.get('nom', 'N/A')} (ID: {quartier.get('id', 'N/A')})")
            
            # Test de sélection d'un quartier
            print(f"\n🧪 TEST DE SÉLECTION DE QUARTIER:")
            if regions and regions[0].get('prefectures'):
                prefecture = regions[0]['prefectures'][0]
                if prefecture.get('communes'):
                    commune = prefecture['communes'][0]
                    if commune.get('quartiers'):
                        quartier = commune['quartiers'][0]
                        quartier_id = quartier.get('id')
                        print(f"   - Quartier sélectionné: {quartier.get('nom')} (ID: {quartier_id})")
                        print(f"   - ✅ Quartier valide pour l'inscription")
                    else:
                        print(f"   - ❌ Aucun quartier trouvé")
                else:
                    print(f"   - ❌ Aucune commune trouvée")
            else:
                print(f"   - ❌ Aucune préfecture trouvée")
            
            print(f"\n🎯 RÉSUMÉ:")
            print(f"   - API géographique: ✅ Fonctionnelle")
            print(f"   - Données disponibles: ✅ {total_quartiers} quartiers")
            print(f"   - Structure: ✅ Valide")
            
        else:
            print(f"❌ Erreur: {response.status_code}")
            print(f"Réponse: {response.text}")
            
    except Exception as e:
        print(f"❌ Erreur lors du test: {e}")

if __name__ == "__main__":
    test_geographic_data() 

File: test_geographic_data.js
// Script de test pour diagnostiquer le problème des données géographiques
const axios = require('axios');

// Configuration
const API_BASE_URL = 'http://localhost:8000/api';
const FRONTEND_API_URL = 'https://communiconnect-backend.onrender.com/api';

async function testGeographicData() {
  console.log('🔍 Test des données géographiques...\n');

  // Test 1: API locale
  console.log('1️⃣ Test API locale (localhost:8000):');
  try {
    const localResponse = await axios.get(`${API_BASE_URL}/users/geographic-data/`);
    console.log('✅ API locale fonctionne');
    console.log(`📊 Données reçues: ${localResponse.data.regions?.length || 0} régions`);
    console.log(`📊 Premier quartier: ${localResponse.data.regions?.[0]?.prefectures?.[0]?.communes?.[0]?.quartiers?.[0]?.nom || 'Aucun'}`);
  } catch (error) {
    console.log('❌ Erreur API locale:', error.message);
  }

  console.log('\n2️⃣ Test API de production (Render):');
  try {
    const prodResponse = await axios.get(`${FRONTEND_API_URL}/users/geographic-data/`);
    console.log('✅ API de production fonctionne');
    console.log(`📊 Données reçues: ${prodResponse.data.regions?.length || 0} régions`);
  } catch (error) {
    console.log('❌ Erreur API de production:', error.message);
  }

  // Test 3: Configuration frontend
  console.log('\n3️⃣ Test configuration frontend:');
  console.log(`🔗 URL API configurée: ${process.env.REACT_APP_API_URL || 'Non définie'}`);
  console.log(`🔗 URL par défaut: ${FRONTEND_API_URL}`);

  // Test 4: Simulation du composant GeographicSelector
  console.log('\n4️⃣ Test simulation GeographicSelector:');
  try {
    const testData = await axios.get(`${API_BASE_URL}/users/geographic-data/`);
    const regions = testData.data.regions;
    
    if (regions && regions.length > 0) {
      const firstRegion = regions[0];
      const firstPrefecture = firstRegion.prefectures?.[0];
      const firstCommune = firstPrefecture?.communes?.[0];
      const firstQuartier = firstCommune?.quartiers?.[0];
      
      console.log('✅ Données disponibles:');
      console.log(`   Région: ${firstRegion.nom}`);
      console.log(`   Préfecture: ${firstPrefecture?.nom}`);
      console.log(`   Commune: ${firstCommune?.nom}`);
      console.log(`   Quartier: ${firstQuartier?.nom}`);
      
      // Test de sélection
      const selection = {
        region_id: firstRegion.id,
        prefecture_id: firstPrefecture?.id,
        commune_id: firstCommune?.id,
        quartier_id: firstQuartier?.id
      };
      
      console.log('\n📝 Sélection testée:', selection);
    } else {
      console.log('❌ Aucune donnée géographique disponible');
    }
  } catch (error) {
    console.log('❌ Erreur lors du test de sélection:', error.message);
  }

  // Test 5: Vérification des CORS
  console.log('\n5️⃣ Test CORS:');
  try {
    const corsResponse = await axios.options(`${API_BASE_URL}/users/geographic-data/`);
    console.log('✅ CORS configuré correctement');
    console.log('📋 Headers CORS:', corsResponse.headers);
  } catch (error) {
    console.log('❌ Problème CORS:', error.message);
  }

  console.log('\n🎯 Recommandations:');
  console.log('1. Vérifiez que le backend est démarré sur le port 8000');
  console.log('2. Vérifiez la configuration CORS dans Django');
  console.log('3. Vérifiez les variables d\'environnement du frontend');
  console.log('4. Testez l\'API directement dans le navigateur');
}

// Exécution du test
testGeographicData().catch(console.error); 